shell/platform/linux/fl_platform_plugin.cc

```diff
@@ -0,0 +1,172 @@
+    return;
+  }
+
+  g_autoptr(FlValue) result = fl_value_new_map();
+  fl_value_set_string_take(result, kTextKey, fl_value_new_string(text));
+
+  g_autoptr(GError) error = nullptr;
+  if (!fl_method_call_respond_success(method_call, result, &error))
+    g_warning("Failed to send method call response: %s", error->message);
+}
+
+// Called when Flutter wants to copy to the clipboard.
+static FlMethodResponse* clipboard_set_data(FlPlatformPlugin* self,
+                                            FlValue* args) {
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_MAP) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Argument map missing or malformed", nullptr));
+  }
+
+  FlValue* text_value = fl_value_lookup_string(args, kTextKey);
+  if (text_value == nullptr ||
+      fl_value_get_type(text_value) != FL_VALUE_TYPE_STRING) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing clipboard text", nullptr));
+  }
+
+  GtkClipboard* clipboard =
+      gtk_clipboard_get_default(gdk_display_get_default());
+  gtk_clipboard_set_text(clipboard, fl_value_get_string(text_value), -1);
+
+  return FL_METHOD_RESPONSE(fl_method_success_response_new(nullptr));
+}
+
+// Called when Flutter wants to paste from the clipboard.
+static void clipboard_get_data_async(FlPlatformPlugin* self,
+                                     FlMethodCall* method_call) {
+  FlValue* args = fl_method_call_get_args(method_call);
+
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_STRING) {
+    g_autoptr(GError) error = nullptr;
+    if (!fl_method_call_respond_error(method_call, kBadArgumentsError,
+                                      "Expected string", nullptr, &error))
+      g_warning("Failed to send method call response: %s", error->message);
+    return;
+  }
+
+  const gchar* format = fl_value_get_string(args);
+  if (strcmp(format, kTextPlainFormat) != 0) {
+    g_autoptr(GError) error = nullptr;
+    if (!fl_method_call_respond_error(method_call, kUnknownClipboardFormatError,
+                                      "GTK clipboard API only supports text.",
+                                      nullptr, &error))
+      g_warning("Failed to send method call response: %s", error->message);
+    return;
+  }
+
+  GtkClipboard* clipboard =
+      gtk_clipboard_get_default(gdk_display_get_default());
+  gtk_clipboard_request_text(clipboard, clipboard_text_cb,
+                             g_object_ref(method_call));
+}
+
+// Called when Flutter wants to quit the application.
+static FlMethodResponse* system_navigator_pop(FlPlatformPlugin* self) {
+  GApplication* app = g_application_get_default();
+  if (app == nullptr) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kFailedError, "Unable to get GApplication", nullptr));
+  }
+
+  g_application_quit(app);
+
+  return FL_METHOD_RESPONSE(fl_method_not_implemented_response_new());
```

Comment:
Why not implemented rather than success? I know it doesn't really matter given that we're quitting, but if you have to have a return value why not the right one?

Reply:
Forgot to update this when I implemented it.